presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionTask.java

```diff
@@ -136,138 +106,112 @@ public NativeExecutionTask(
                     .keySet().stream()
                     .map(outputId -> new BufferInfo(outputId, false, 0, 0, PageBufferInfo.empty()))
                     .collect(toImmutableList());
-            this.taskInfo.set(createInitialTask(taskId, URI.create(LOCAL_URI + taskId), bufferStates, new TaskStats(DateTime.now(), null), ""));
-            this.taskUpdateRequestCodec = taskUpdateRequestCodec;
-            this.planFragmentCodec = planFragmentCodec;
-            this.taskInfoCodec = taskInfoCodec;
+            this.workerClient = new PrestoSparkHttpWorkerClient(httpClient, taskId, URI.create(LOCAL_URI));
```

Comment:
Do we want to make the `URI` configurable from the user since we could pass in different port as part of the URI for example?

Reply:
Good point!